In [3]:
import pandas as pd
from sklearn.model_selection import train_test_split
from sklearn.linear_model import LogisticRegression
from sklearn.metrics import accuracy_score

mushroom_data = pd.read_csv("mushroom_data_preprocessed.csv")

In [4]:
odor_dummies = pd.get_dummies(mushroom_data["odor"], prefix="odor")
spore_color_dummies = pd.get_dummies(mushroom_data["stalk-surface-above-ring"], prefix="stalk-surface-above-ring")


In [5]:
predictors = pd.concat([odor_dummies, spore_color_dummies], axis=1)


In [11]:
predictors.head()

odor_a  odor_c  odor_f  odor_l  odor_m  odor_n  odor_p  odor_s  odor_y  \
0       0       0       0       0       0       0       1       0       0   
1       1       0       0       0       0       0       0       0       0   
2       0       0       0       1       0       0       0       0       0   
3       0       0       0       0       0       0       1       0       0   
4       0       0       0       0       0       1       0       0       0   

   stalk-surface-above-ring_d  stalk-surface-above-ring_g  \
0                           0                           0   
1                           0                           1   
2                           0                           0   
3                           0                           0   
4                           0                           1   

   stalk-surface-above-ring_l  stalk-surface-above-ring_m  \
0                           0                           0   
1                           0                           0   
2                           0                           1   
3                           0                           0   
4                           0                           0   

   stalk-surface-above-ring_p  stalk-surface-above-ring_u  \
0                           0                           1   
1                           0                           0   
2                           0                           0   
3                           0                           1   
4                           0                           0   

   stalk-surface-above-ring_w  
0                           0  
1                           0  
2                           0  
3                           0  
4                           0

In [8]:
X_train, X_test, y_train, y_test = train_test_split(predictors, mushroom_data["edible"], test_size=0.3, random_state=42)
lr = LogisticRegression()
lr.fit(X_train, y_train)


LogisticRegression()

In [9]:
y_pred = lr.predict(X_test)
accuracy = accuracy_score(y_test, y_pred)
print(f"Accuracy: {accuracy}")


Accuracy: 0.9840032813781788


In [10]:
from sklearn.metrics import confusion_matrix

confusion_matrix = confusion_matrix(y_test, y_pred)
print(confusion_matrix)


[[1257    0]
 [  39 1142]]


Based on the confusion matrix that you provided, it appears that the logistic regression model correctly classified all 1257 edible mushrooms and correctly classified 1142 poisonous mushrooms, which is impressive. However, it incorrectly classified 39 poisonous mushrooms as edible and did not correctly classify any edible mushrooms as poisonous